reproducibility_project/init.py

```diff
@@ -130,87 +130,80 @@ def dict_product(dd):
-                                                        ),
-                                                        decimals=3,
-                                                    ),
-                                                    "init_vap_den": np.round(
-                                                        init_vap_den.to_value(
-                                                            g_per_cm3
-                                                        ),
-                                                        decimals=3,
-                                                    )
-                                                    if init_vap_den
-                                                    else None,
-                                                    "mass": np.round(
-                                                        mass.to_value("amu"),
-                                                        decimals=3,
-                                                    ),
-                                                    "forcefield_name": forcefields[
-                                                        molecule
-                                                    ],
-                                                    "cutoff_style": cutoff_style,
-                                                    "r_cut": np.round(
-                                                        r_cuts[
-                                                            molecule
-                                                        ].to_value("nm"),
-                                                        decimals=3,
-                                                    ),
-                                                }
-                                                total_statepoints.append(
-                                                    statepoint
-                                                )
+    for (
+        engine,
+        ensemble,
+        (temp, press),
+        n_liq,
+        liq_box_L,
+        n_vap,
+        vap_box_L,
+        (init_liq_den, init_vap_den),
+        mass,
+        cutoff_style,
+        replica,
+    ) in itertools.product(
+        simulation_engines,
+        ensembles[molecule],
+        zip(temperatures[molecule], pressures[molecule]),
+        N_liq_molecules[molecule],
+        liq_box_lengths[molecule],
+        N_vap_molecules[molecule],
+        vap_box_lengths[molecule],
+        zip(init_density_liq[molecule], init_density_vap[molecule]),
+        masses[molecule],
+        cutoff_styles,
+        replicas,
+    ):
+        statepoint = {
+            "molecule": molecule,
+            "engine": engine,
+            "replica": replica,
+            "temperature": np.round(
+                temp.to_value("K"),
+                decimals=3,
+            ),
+            "pressure": np.round(
+                press.to_value("kPa"),
+                decimals=3,
+            ),
```

Comment:
```suggestion
            "pressure": np.round(press.to_value("kPa"), decimals=3).item(),
```
if we want to silence the signac/numpy warning, we could do this